src/models/tiles/tile-model.ts

```diff
@@ -151,6 +149,30 @@ export const TileModel = types
     }
   }));
 
+// The types.snapshotProcessor is used here so the types are handled better
+// It updates the TileModel.create(...) parameter type and the
+// SnapshotIn type to match the type of parameter of preProcessor.
+// It also updates the SnapshotOut type to be the return type of postProcessor
+// I didn't test extensively perhaps there is a way to do this with the alternative approach
+export const TileModel = types.snapshotProcessor(TileModelInternal, {
+  preProcessor(snapshot: Omit<SnapshotIn<typeof TileModelInternal>, "internalTitle"> & {title?: string}) {
+    // internally we store the title in the `internalTitle` property
+    // this way we can provide view that computes the title
+    // when serialized we save it out simply as `title`
+    const {title, ...others} = clone(snapshot);
```

Comment:
I don't think you need to `clone` here. The destructuring will create a new object automatically. The same is true below in the `postProcessor`.

Reply:
True. Good catch. I was being paranoid about not modifying the snapshot in place, but you're right the clone isn't needed.